src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -360,12 +375,39 @@ private bool TryHandleAsKeySegment(string segmentText)
             KeySegment previousKeySegment = this.FindPreviousKeySegment();
 
             KeySegment keySegment;
+
+            bool trybindingEscapeFunction = false;
+
+            if (segmentText.Length > 0 && segmentText[segmentText.Length - 1] == ':' && previous.EdmType != null)

```

Comment:
>segmentText.Length > 0 && [](start = 16, length = 25)

is this ever called with a zero-length segmentText, or should this just be an assert?